### PR TITLE
[v7r0] Fix Profiler for CPU estimate

### PR DESCRIPTION
**Pull request by phicharp**
BEGINRELEASENOTES

*Core
FIX: The CPU for the jobWrapper process group was incorrectly calculated as it was not including former children of the JobWrapper.

ENDRELEASENOTES

In addition fix a few modules for preparing for python3 (remove .keys(), .iterxxx() -> xxx(), basestring ->six.string_types...)

### CODE REVIEW

**Comment by phicharp** [comment]
Should I fix all the failing tests?

**Comment by fstagni** [comment]
I guess you first opened the PR against the integration branch, right? That's why of the failing pep8 test.

**Comment by phicharp** [comment]
Ah yes, that was my usual mistake, I didn't check, but now it should be OK, right? Should I push another commit to trigger another set of tests?

File: DataManagementSystem/Client/DataManager.py [hunk 1]
```diff
@@ -194,7 +195,7 @@ def __cleanDirectory(self, folder):
     res = self.removeFile(res['Value'])
     if not res['OK']:
       return res
-    for lfn, reason in res['Value']['Failed'].iteritems():
+    for lfn, reason in res['Value']['Failed'].items():
```
[the remark this answers is not in the document]
Comment (phicharp) [reply DataManagementSystem/Client/DataManager.py:198]: IIUC all those that used to be `iteritems()` are safe to be set `items()`. We should check the `items()` that must have a `list()`, right... I'll have a look at these files about this...

[the remark this answers is not in the document]
Comment (phicharp) [reply DataManagementSystem/Client/DataManager.py]: Done

File: DataManagementSystem/Client/DataManager.py [hunk 2]
```diff
@@ -1610,7 +1611,7 @@ def __filterTapeReplicas(self, replicaDict, diskOnly=False):
     The input argument is modified
     """
     seList = set(
-        se for ses in replicaDict['Successful'].itervalues() for se in ses)
+        se for ses in replicaDict['Successful'].values() for se in ses)
```
[the remark this answers is not in the document]
Comment (phicharp) [reply DataManagementSystem/Client/DataManager.py]: Done

**Comment by phicharp** [comment]
Thanks Federico for the approval... although I don't understand why GitHub still says a review is still required and merging is blocked?

**Comment by andresailer** [comment]
In principle, it requires the approval from the code owners. In practice I think the admins can overrule that requirement

**Pull request merged**